test/box-luatest/meta-temporary_spaces_test.lua

```diff
@@ -0,0 +1,440 @@
+
+        local s = box.schema.space.create('metatemp2', { temporary = 'meta',
+                                                         id = 9999 })
+        t.assert_equals(s.temporary, 'meta')
+        s:drop()
+
+        local s = box.schema.space.create('temp', { temporary = true })
+        t.assert_equals(s.temporary, true)
+
+        local s = box.schema.space.create('r1', { temporary = false })
+        t.assert_equals(s.temporary, false)
+        s:drop()
+
+        local s = box.schema.space.create('r2', { temporary = nil })
+        t.assert_equals(s.temporary, false)
+        s:drop()
+
+        t.assert_error_msg_contains(
+            "option 'temporary' can either be true, false or 'meta'",
+            box.schema.space.create, 'super', { temporary = 'super' }
+        )
+
+        t.assert_error_msg_contains(
+            "engine does not support temporary flag",
+            box.schema.space.create,
+                'no-vinyl', { engine = 'vinyl', temporary = 'meta' }
+        )
+
+        local s = box.space.temp
+        t.assert_error_msg_contains(
+            "options parameter 'temporary' should be of type boolean",
+            s.alter, s, { temporary = 'meta' }
+        )
+        t.assert_error_msg_contains(
+            "meta-temporariness cannot change",
+            _space.update,
+                _space, s.id, {{'=', 6, { temporary = 'meta' }}}
+        )
+
+        local s = box.space.metatemp
+        t.assert_error_msg_contains(
+            "meta-temporariness cannot change",
+            s.alter,
+                s, { temporary = true }
+        )
+        t.assert_error_msg_contains(
+            "meta-temporariness cannot change",
+            _space.update,
+                _space, s.id, {{'=', 6, { temporary = true }}}
+        )
+    end)
+end
+
+-- check which features aren't supported for meta-temporary spaces
+g.test_meta_temporary_dont_support = function()
+    g.server:exec(function()
+        local t = require 'luatest'
+
+        local s = box.schema.space.create('metatemp', { temporary = 'meta',
+                                                        id = 1111111111 })
+        s:format {{'k', 'unsigned'}}
+        s:create_index('pk')
+        box.schema.func.create('foo', {
+            is_sandboxed = true,
+            is_deterministic = true,
+            body = "function() end"
+        })
+        t.assert_error_msg_equals(
+            "meta-temporary space does not support functional indexes",
+            s.create_index, s, 'func', { func = 'foo' }
+        )
+
+        t.assert_error_msg_equals(
+            "Grant access to meta-temporary space 'metatemp' "
+            .. "is denied for user 'guest'",
+            box.schema.user.grant, 'guest', 'read', 'space', s.name
+        )
+    end)
+end
+
+-- check which sql features aren't supported for meta-temporary spaces
+g.test_meta_temporary_sql_dont_support = function()
+    g.server:exec(function()
+        local t = require 'luatest'
```

Comment:
`local t = require 'luatest'`. At least in the latest master. You can use the same `t` inside execs as declared globally in the test file.

Reply:
done